owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/capabilities/OCCapabilityViewModel.kt

```diff
@@ -20,81 +20,71 @@
-    private val getStoredCapabilitiesUseCase: GetStoredCapabilitiesUseCase = GetStoredCapabilitiesUseCase(context, account),
-    private val refreshCapabilitiesUseCase: RefreshCapabilitiesUseCase = RefreshCapabilitiesUseCase(context, account),
+    private val accountName: String,
+    getCapabilitiesAsLiveDataUseCase: GetCapabilitiesAsLiveDataUseCase,
+    private val getStoredCapabilitiesUseCase: GetStoredCapabilitiesUseCase,
+    private val refreshCapabilitiesFromServerUseCase: RefreshCapabilitiesFromServerAsyncUseCase,
     private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
 ) : ViewModel() {
 
-    private val _capabilities = MediatorLiveData<UIResult<OCCapabilityEntity>>()
-    val capabilities: LiveData<UIResult<OCCapabilityEntity>> = _capabilities
+    private val _capabilities = MediatorLiveData<UIResult<OCCapability>>()
+    val capabilities: LiveData<UIResult<OCCapability>> = _capabilities
 
-    private var capabilitiesLiveData: LiveData<OCCapabilityEntity>? = getCapabilitiesLiveDataUseCase.execute(
-        GetCapabilitiesLiveDataUseCase.Params(
-            accountName = account.name
+    private var capabilitiesLiveData: LiveData<OCCapability?>? = getCapabilitiesAsLiveDataUseCase.execute(
+        GetCapabilitiesAsLiveDataUseCase.Params(
+            accountName = accountName
         )
-    ).data
+    )
 
     init {
         _capabilities.addSource(capabilitiesLiveData!!) { capabilities ->
-            _capabilities.postValue(UIResult.success(capabilities))
+            _capabilities.postValue(UIResult.Success(capabilities))
         }
 
         refreshCapabilitiesFromNetwork()
     }
 
-    fun getStoredCapabilities(): OCCapabilityEntity? = getStoredCapabilitiesUseCase.execute(
+    fun getStoredCapabilities(): OCCapability? = getStoredCapabilitiesUseCase.execute(
         GetStoredCapabilitiesUseCase.Params(
-            accountName = account.name
+            accountName = accountName
         )
-    ).data
+    )
 
-    private fun refreshCapabilitiesFromNetwork() {
-        viewModelScope.launch(ioDispatcher) {
+    fun refreshCapabilitiesFromNetwork() {
+        viewModelScope.launch {
             _capabilities.postValue(
-                UIResult.loading(capabilitiesLiveData?.value)
+                UIResult.Loading(capabilitiesLiveData?.value)
             )
 
-            val useCaseResult = refreshCapabilitiesUseCase.execute(
-                RefreshCapabilitiesUseCase.Params(
-                    accountName = account.name
+            val useCaseResult = withContext(ioDispatcher) {
+                refreshCapabilitiesFromServerUseCase.execute(
+                    RefreshCapabilitiesFromServerAsyncUseCase.Params(
+                        accountName = accountName
+                    )
                 )
-            )
+            }
 
-            if (!useCaseResult.isSuccess()) {
+            if (!useCaseResult.isSuccess) {
```

Comment:
if more readable if you use isError instead

Reply:
Done